File: eSalon/eSalon.Model/Obavijest.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace eSalon.Model
{
    public class Obavijest
    {
        public int ObavijestId { get; set; }

        public int KorisnikId { get; set; }

        public string Naslov { get; set; } = null!;

        public string Sadrzaj { get; set; } = null!;

        public DateTime DatumObavijesti { get; set; }

        public bool JePogledana { get; set; }

        public string? ImePrezime { get; set; }
    }
}

File: eSalon/UI/esalon_desktop/lib/models/obavijest.dart
import 'package:json_annotation/json_annotation.dart';

part 'obavijest.g.dart';

@JsonSerializable()
class Obavijest {
  int obavijestId;
  int korisnikId;
  String naslov;
  String sadrzaj;
  DateTime datumObavijesti;
  bool jePogledana;
  String? imePrezime;

  Obavijest(
    this.obavijestId,
    this.korisnikId,
    this.naslov,
    this.sadrzaj,
    this.datumObavijesti,
    this.jePogledana,
    this.imePrezime,
  );

  factory Obavijest.fromJson(Map<String, dynamic> json) => _$ObavijestFromJson(json);

  Map<String, dynamic> toJson() => _$ObavijestToJson(this);
}
